admin/apple-actions/index/class-export.php

```diff
@@ -115,6 +115,21 @@ public function fetch_exporter() {
 		// Get the content.
 		$content = $this->get_content( $post );
 
+		/*
+		 * If the excerpt looks too similar to the content, remove it.
+		 * We do this before the filter, to allow overrides for the final value.
+		 * This essentially prevents the case where someone intentionally copies
+		 * the first paragraph of content into the `post_excerpt` field and
+		 * unintentionally introduces a duplicate content issue.
+		 */
+		if ( ! empty( $excerpt ) ) {
+			$content_normalized = strtolower( str_replace( ' ', '', wp_strip_all_tags( $content ) ) );
+			$excerpt_normalized = strtolower( str_replace( ' ', '', wp_strip_all_tags( $excerpt ) ) );
+			if ( false !== strpos( $content_normalized, $excerpt_normalized ) ) {
```

Comment:
Won't this logic remove the excerpt no matter where it appears in the content, rather than just if it's identical to the start of the content? Is that expected? For example, if lede of a story is an anecdote, and the nut graf is a few paragraphs down, and the user copies the nut graf for the excerpt?

Reply:
Yes, it will, and yes, that's expected. Apple reviews articles from channels before approval, and one of the things they check is whether an article contains "repeating text," although there isn't specific guidance around what exactly trips that alarm, so this fix ensures that if the entirety of the excerpt appears somewhere in the article, we're not including that in the Intro component that appears before the article automatically. The Intro component should be used to introduce the article, rather than copying parts of it, so while the example you gave would be a good use case for a custom excerpt to appear in search results and recirc around the site, it wouldn't (imo) make a good Intro component in Apple News. We're also doing this logic before the filter to explicitly set the excerpt, so if a user doesn't like this default behavior, they can set it back via the filter.